### ISSUE TITLE
spontaneous daily streak reset, dailies completed at startup, incorrect Perfect Day/Streak counts

### ISSUE THREAD

**Issue by chaoticawesome** [HabitRPG/habitica#2768]
I logged in this morning and I got a notification that I had leveled up, along with about 25 gold and 16 mana instead of the usual 10. My dailies were all checked off and the streaks had all been reset to 2. My habits hadn't aged properly - I do them the minimum number of times to make them bright blue, so usually they're teal the next day, and they were still bright blue while teal tasks had turned green like normal. I was also lacking my Perfect Day buff, despite having completed all of my dailies yesterday.
![habitrpgproblem1](https://f.cloud.github.com/assets/6465482/2101604/c37547ac-8f54-11e3-8d4f-7c23d49f7849.png)
Checking the score reveals that I earned nothing yesterday, even though I completed my dailies, kept up my habits and did two to-dos.
![habitrpgproblem2](https://f.cloud.github.com/assets/6465482/2101671/42cf527c-8f55-11e3-8834-ef44ebca4f1d.png)
In addition, my Perfect Day and 21-Day Streak counter was wrong - I had 5 streaks and 29 perfect days as of yesterday.
![habitrpgproblem3](https://f.cloud.github.com/assets/6465482/2101762/1a08b328-8f56-11e3-81f3-caee8a157d69.png)
I apparently don't anymore.

At this point I'm thinking the game may have somehow skipped me forward a day - since I leveled up on login I don't know if I had taken damage from "missed dailies". It doesn't explain the erroneous Perfect Day/Streak counter or the strangely behaving habits but it explains everything else. This sadly isn't the first time I've had issues with HabitRPG, and I hope it gets fixed because I do enjoy using your app!

Browser: Firefox 26.0
User ID: 79167200-f69d-4af3-bcac-6d166c21fd31

**Comment by colegleason** [comment]
Just wondering, how do you have 29 perfect days when the Perfect Day
feature was added less than 29 days ago?

On Thu, Feb 6, 2014 at 12:06 PM, chaoticawesome notifications@github.comwrote:

> I logged in this morning and I got a notification that I had leveled up,
> along with about 25 gold and 16 mana instead of the usual 10. My dailies
> were all checked off and the streaks had all been reset to 2. My habits
> hadn't aged properly - I do them the minimum number of times to make them
> bright blue, so usually they're teal the next day, and they were still
> bright blue while teal tasks had turned green like normal. I was also
> lacking my Perfect Day buff, despite having completed all of my dailies
> yesterday.
> [image: habitrpgproblem1]https://f.cloud.github.com/assets/6465482/2101604/c37547ac-8f54-11e3-8d4f-7c23d49f7849.png
> Checking the score reveals that I earned nothing yesterday, even though I
> completed my dailies, kept up my habits and did two to-dos.
> [image: habitrpgproblem2]https://f.cloud.github.com/assets/6465482/2101671/42cf527c-8f55-11e3-8834-ef44ebca4f1d.png
> In addition, my Perfect Day and 21-Day Streak counter was wrong - I had 5
> streaks and 29 perfect days as of yesterday.
> [image: habitrpgproblem3]https://f.cloud.github.com/assets/6465482/2101762/1a08b328-8f56-11e3-81f3-caee8a157d69.png
> I apparently don't anymore.
> 
> At this point I'm thinking the game may have somehow skipped me forward a
> day - since I leveled up on login I don't know if I had taken damage from
> "missed dailies". It doesn't explain the erroneous Perfect Day/Streak
> counter or the strangely behaving habits but it explains everything else.
> This sadly isn't the first time I've had issues with HabitRPG, and I hope
> it gets fixed because I do enjoy using your app!
> 
> Browser: Firefox 26.0
> User ID: 79167200-f69d-4af3-bcac-6d166c21fd31
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/HabitRPG/habitrpg/issues/2768
> .

## 

Cole Gleason
Student, University of Illinois at Urbana-Champaign
Computer Science, Class of 2015
cg@colegleason.com | colegleason.com

**Comment by chaoticawesome** [comment]
I don't remember when it was added but it should be higher than that? My highest streaks were at 29 days.

**Comment by deilann** [comment]
Did you lose any health?

**Comment by chaoticawesome** [comment]
"since I leveled up on login I don't know if I had taken damage from "missed dailies"."

I legitimately don't know. I was at full health when I stopped last night, and while I didn't get a notification telling me I got health back I also didn't get a notification of how much exp I "earned" when it checked off all my dailies. It also counted all of them at once and gave me a single notification for how much gold I earned for all of them.
